File: Práctica Grupal 1/src/gestorAplicacion/paquete1/Libro.java
package gestorAplicacion.paquete1;
import gestorAplicacion.paquete2.Autor;

import java.util.ArrayList;
import java.io.Serializable;

public class Libro extends Recurso implements Serializable {
	private static final long serialVersionUID = 1L;
	static protected int totalLibros;
    protected String isbn;
    protected Autor autor;
    protected int año;

    // Constructor de la clase Libro
    public Libro(String nombre, int idRecurso, String isbn, Autor autor, int año) {
        super(nombre, idRecurso); // Llama al constructor de la clase base (Recurso)
        this.isbn = isbn;
        this.autor = autor;
        this.año = año;
        autor.getObras().add(this);
        totalLibros++;
    }
    
    public Libro() {
    	this("Libro sin nombre", 0, "",new Autor(),0);
    }
    

    // Métodos para obtener el ISBN del libro
    public String getIsbn() {
        return isbn;
    }

    // Métodos para establecer el ISBN del libro
    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    // Métodos para obtener el autor del libro
    public Autor getAutor() {
        return autor;
    }

    // Métodos para establecer el autor del libro
    public void setAutor(Autor autor) {
        this.autor = autor;
    }

    // Métodos para obtener el año de publicación del libro
    public int getAño() {
        return año;
    }
    
    public static int getTotalLibros() {
		return totalLibros;
	}

    // Métodos para establecer el año de publicación del libro
    public void setAño(int año) {
        this.año = año;
    }
    
    public String tipoRecurso() {
    	return "Libro";
    }
    
    public String toString() {
    	return this.getNombre();
    }
}

File: Práctica Grupal 2/practica2-grupo1-Equipo2/src/gestorAplicacion/clasesDeBiblioteca/Libro.py
from .Recurso import Recurso

class Libro(Recurso):
    totalLibros = 0

    def __init__(self, nombre, idRecurso, isbn, autor, año):
        super().__init__(nombre, idRecurso, "Libro")
        self.isbn = isbn
        self.autor = autor
        self.año = año
        autor.get_obras().append(self)
        Libro.totalLibros += 1
        self.idRecurso = Libro.totalLibros
    
    def get_nombre(self):
        return self.nombre
    
    def get_isbn(self):
        return self.isbn

    def set_isbn(self, isbn):
        self.isbn = isbn

    def get_autor(self):
        return self.autor

    def set_autor(self, autor):
        self.autor = autor

    def get_año(self):
        return self.año

    def set_año(self, año):
        self.año = año

    def tipo_recurso(self):
        return "Libro"

    def __str__(self):
        return self.get_nombre()

    @staticmethod
    def get_total_libros():
        return Libro.totalLibros
